tests/tests_geomstats/test_full_rank_correlation_matrices.py

```diff
@@ -1,148 +1,237 @@
-
-    def test_horizontal_lift_and_tangent_riemannian_submersion(self):
-        mat = self.base.random_point()
-        vec = self.base.random_point()
-        tangent_vec = self.base.to_tangent(vec, mat)
-        horizontal = self.bundle.horizontal_lift(tangent_vec, base_point=mat)
-        result = self.bundle.tangent_riemannian_submersion(horizontal, mat)
+        self.assertAllClose(is_horizontal, gs.array(True))
+
+    def test_horizontal_lift_is_horizontal(self, n, tangent_vec, mat):
+        bundle = self.space(n)
+        lift = bundle.horizontal_lift(gs.array(tangent_vec), gs.array(mat))
+        result = gs.all(bundle.is_horizontal(lift, gs.array(mat)))
+        self.assertAllClose(result, gs.array(True))
+
+    def test_vertical_projection_is_vertical(self, n, tangent_vec, mat):
+        bundle = self.space(n)
+        proj = bundle.vertical_projection(gs.array(tangent_vec), gs.array(mat))
+        result = gs.all(bundle.is_vertical(proj, gs.array(mat)))
+        self.assertAllClose(result, gs.array(True))
+
+    @autograd_tf_and_torch_only
+    def test_align_is_horizontal(self, n, point_a, point_b):
+        bundle = self.space(n)
+        aligned = bundle.align(point_a, point_b, tol=1e-10)
+        log = bundle.ambient_metric.log(aligned, point_b)
+        result = bundle.is_horizontal(log, point_b, atol=gs.atol * 10000)
+        self.assertAllClose(result, gs.array(True))
+
+    def test_horizontal_lift_and_tangent_riemannian_submersion(
+        self, n, tangent_vec, mat
+    ):
+        bundle = self.space(n)
+        horizontal = bundle.horizontal_lift(gs.array(tangent_vec), gs.array(mat))
+        result = bundle.tangent_riemannian_submersion(horizontal, gs.array(mat))
         self.assertAllClose(result, tangent_vec)
 
-    def test_is_horizontal(self):
-        mat = self.base.random_point()
-        vec = self.base.random_point()
-        tangent_vec = self.base.to_tangent(vec, mat)
-        horizontal = self.bundle.horizontal_lift(tangent_vec, mat)
-        result = self.bundle.is_horizontal(horizontal, mat)
-        self.assertTrue(result)
-
-    def test_is_vertical(self):
-        mat = self.bundle.random_point()
-        tangent_vec = self.bundle.random_point()
-        vertical = self.bundle.vertical_projection(tangent_vec, mat)
-        result = self.bundle.is_vertical(vertical, mat)
-        self.assertTrue(result)
-
-    def test_inner_product(self):
-        def inner_prod(tangent_vec_a, tangent_vec_b, base_point):
-            affine_part = self.bundle.ambient_metric.inner_product(
-                tangent_vec_a, tangent_vec_b, base_point
-            )
-            n = tangent_vec_b.shape[-1]
 
-            inverse_base_point = GeneralLinear.inverse(base_point)
-            operator = gs.eye(n) + base_point * inverse_base_point
-            inverse_operator = GeneralLinear.inverse(operator)
+class TestFullRankCorrelationAffineQuotientMetric(TestCase, metaclass=Parametrizer):
```

Comment:
why not RiemannianMetricParametrizer?

Reply:
Note for improvements: ultimately, it should have QuotientMetricParametrizer as metaclass

Reply:
Tests are failing. (I explained this in standup texts). For time being this is being skipped